cpp/open3d/t/pipelines/kernel/ComputeTransformCUDA.cu

```diff
@@ -43,86 +96,24 @@ void ComputePosePointToPlaneCUDA(const float *source_points_ptr,
-                        3 * correspondences_first[workload_idx];
-                const int64_t &target_idx =
-                        3 * correspondences_second[workload_idx];
-
-                const int64_t atai_stride = 21 * workload_idx;
-                const int64_t atbi_stride = 6 * workload_idx;
-
-                const float &sx = source_points_ptr[source_idx + 0];
-                const float &sy = source_points_ptr[source_idx + 1];
-                const float &sz = source_points_ptr[source_idx + 2];
-                const float &tx = target_points_ptr[target_idx + 0];
-                const float &ty = target_points_ptr[target_idx + 1];
-                const float &tz = target_points_ptr[target_idx + 2];
-                const float &nx = target_normals_ptr[target_idx + 0];
-                const float &ny = target_normals_ptr[target_idx + 1];
-                const float &nz = target_normals_ptr[target_idx + 2];
-
-                const float bi =
-                        (tx - sx) * nx + (ty - sy) * ny + (tz - sz) * nz;
-                const float ai[] = {(nz * sy - ny * sz),
-                                    (nx * sz - nz * sx),
-                                    (ny * sx - nx * sy),
-                                    nx,
-                                    ny,
-                                    nz};
-
-                for (int i = 0, j = 0; j < 6; j++) {
-                    for (int k = 0; k <= j; k++) {
-                        atai_ptr[atai_stride + i] = ai[j] * ai[k];
-                        i++;
-                    }
-                    atbi_ptr[atbi_stride + j] = ai[j] * bi;
-                }
-            });
-
-    // Reduce matrix atai (to 1x21) and atbi (to ATB.T() 1x6).
-    // Compute linear system on CPU as Float64.
-    core::Device host("CPU:0");
-    core::Tensor ata_1x21 = atai.Sum({0}, true).To(host, core::Dtype::Float64);
-    core::Tensor ATB = atbi.Sum({0}, true).T().To(host, core::Dtype::Float64);
-
-    //   ata_1x21 is a {1,21} vector having elements of the matrix ATA such
-    //     that the corresponding elemetes in ATA are like:
-    //     0
-    //     1   2
-    //     3   4   5
-    //     6   7   8   9
-    //     10  11  12  13  14
-    //     15  16  17  18  19  20
-    //     Since, ATA is a symmertric matrix, it can be regenerated from this.
-
-    core::Tensor ATA = core::Tensor::Empty({6, 6}, core::Dtype::Float64, host);
-    double *ATA_ptr = ATA.GetDataPtr<double>();
-    double *ata_1x21_ptr = ata_1x21.GetDataPtr<double>();
-
-    for (int i = 0, j = 0; j < 6; j++) {
-        for (int k = 0; k <= j; k++) {
-            ATA_ptr[j * 6 + k] = ata_1x21_ptr[i];
-            ATA_ptr[k * 6 + j] = ata_1x21_ptr[i];
-            i++;
-        }
-    }
+    core::Tensor global_sum =
+            core::Tensor::Zeros({29}, core::Dtype::Float32, device);
+    float *global_sum_ptr = global_sum.GetDataPtr<float>();
+
+    const int THREAD_1D_UNIT = 256;
```

Comment:
Better to claim it globally, also maybe follow the Google style kBlockSize or kThread1DUnit

Reply:
Done.